### ISSUE TITLE
"eth1 node using incorrect chain id, 0 != 1" when geth is not fully synced

### ISSUE THREAD

**Issue by victorelec14** [prysmaticlabs/prysm#7933]
# 🐞 Bug Report

### Description

 
beacon-chain does not start with a private node ETH1 (Docker)

### Has this worked before in a previous version?

 i dont know 

## 🔥 Error
>time="2020-11-23 23:31:55" level=error msg="Could not connect to powchain endpoint" error="could not dial eth1 nodes: eth1 node using incorrect chain id, 0 != 1" prefix=powchain

<pre>
time="2020-11-23 23:31:54" level=warning msg="Running on ETH2 Mainnet" prefix=flags
time="2020-11-23 23:31:54" level=info msg="Using "max_cover" strategy on attestation aggregation" prefix=flags
time="2020-11-23 23:31:54" level=info msg="Checking DB" database-path="/data/beaconchaindata" prefix=node
time="2020-11-23 23:31:55" level=warning msg="No bootstrap addresses supplied" prefix=p2p
time="2020-11-23 23:31:55" level=info msg="Deposit contract: 0x00000000219ab540356cbb839cbe05303d7705fa" prefix=node
time="2020-11-23 23:31:55" level=info msg="Waiting for state to be initialized" prefix=initial-sync
time="2020-11-23 23:31:55" level=info msg="Starting beacon node" prefix=node version="Prysm/v1.0.0-beta.2/f361450e8da18033f5458b018e3e6784295f50ca. Built at: 2020-11-14 02:35:28+00:00"
time="2020-11-23 23:31:55" level=info msg="Waiting to reach the validator deposit threshold to start the beacon chain..." prefix=blockchain
time="2020-11-23 23:31:55" level=info msg="gRPC server listening on port" address="127.0.0.1:4000" prefix=rpc
time="2020-11-23 23:31:55" level=warning msg="You are using an insecure gRPC server. If you are running your beacon node and validator on the same machines, you can ignore this message. If you want to know how to enable secure connections, see: https://docs.prylabs.network/docs/prysm-usage/secure-grpc" prefix=rpc
time="2020-11-23 23:31:55" level=info msg="Starting JSON-HTTP API" address="127.0.0.1:3500" prefix=gateway
time="2020-11-23 23:31:55" level=error msg="Could not connect to powchain endpoint" error="could not dial eth1 nodes: eth1 node using incorrect chain id, 0 != 1" prefix=powchain

<code>

</code></pre>

## 🌍  Your Environment

**Operating System:**

<pre>
  <code>
   docker -v
Docker version 19.03.8, build afacb8b7f0

  </code>
</pre>

**Eth1 Node: (Docker):**

<pre>
  <code>
docker run -p 127.0.0.1:8545:8545 -p 30303:30303 -v $(HOME)/eth2/node:/root ethereum/client-go --http -http.addr "0.0.0.0"

--or-
docker run -p 0.0.0.0:8545:8545 -p 30303:30303 -v $(HOME)/eth2/node:/root ethereum/client-go --http -http.addr "0.0.0.0"

  </code>
</pre>

**Beacon: (Docker):**

<pre>
  <code>

docker run -it -v$(HOME)/eth2/beacon/.eth2:/data -p 4000:4000 -p 13000:13000 -p 12000:12000/udp --name beacon-node \
  gcr.io/prysmaticlabs/prysm/beacon-chain:stable \
  --datadir=/data \
  --rpc-host=127.0.0.1 \
  --http-web3provider="http://192.168.178.170:8545"

  </code>
</pre>

**What version of Prysm are you running? (Which release)**

<pre>
  <code>
gcr.io/prysmaticlabs/prysm/beacon-chain:stable
Prysm/v1.0.0-beta.2/f361450e8da18033f5458b018e3e6784295f50ca
  </code>
</pre>

**Anything else relevant (validator index / public key)?**

**Comment by BiggT76** [comment]
I am also experiencing this problem.

**Comment by victorelec14** [comment]
I have tried with "--mainnet"  which was added in **v1.0.0-beta.2** but it does not work either, it is as if I always want to start in the testnet and eth1 is in mainnet.

With prymont and medal I have had no problems being testnets.

Cleardb too nothing
> --clear-db

**Comment by BiggT76** [comment]
I've heard others having this problem, if I had to guess its because Prysm isn't ready for mainnet yet.  Given all the big bold warnings on their website _not to run_ prysm on mainnet yet...

Hopefully someone "in the know" can offer more insight.

**Comment by prestonvanloon** [comment]
Can you try with RPC enabled in geth? `--rpc`

**Comment by victorelec14** [comment]
> Can you try with RPC enabled in geth? `--rpc`

--http -http.addr are the new parameter, rpc be deprecated in future, but i get the same response with --rpc

**Comment by victorelec14** [comment]
> --verbosity=debug

```
time="2020-11-24 22:14:53" level=warning msg="Running on ETH2 Mainnet" prefix=flags
time="2020-11-24 22:14:53" level=info msg="Using "max_cover" strategy on attestation aggregation" prefix=flags
time="2020-11-24 22:14:53" level=info msg="Checking DB" database-path="/data/beaconchaindata" prefix=node
time="2020-11-24 22:14:54" level=warning msg="No bootstrap addresses supplied" prefix=p2p
time="2020-11-24 22:14:54" level=debug msg="ECDSA private key generated" prefix=p2p
time="2020-11-24 22:14:54" level=info msg="Deposit contract: 0x00000000219ab540356cbb839cbe05303d7705fa" prefix=node
time="2020-11-24 22:14:54" level=info msg="Waiting for state to be initialized" prefix=initial-sync
time="2020-11-24 22:14:54" level=info msg="Starting beacon node" prefix=node version="Prysm/v1.0.0-beta.2/f361450e8da18033f5458b018e3e6784295f50ca. Built at: 2020-11-14 02:35:28+00:00"
time="2020-11-24 22:14:54" level=debug msg="Starting 9 services: [*p2p.Service *powchain.Service *attestations.Service *blockchain.Service *initialsync.Service *sync.Service *rpc.Service *gateway.Gateway *prometheus.Service]" prefix=registry
time="2020-11-24 22:14:54" level=debug msg="Starting service type *p2p.Service" prefix=registry
time="2020-11-24 22:14:54" level=debug msg="Starting service type *powchain.Service" prefix=registry
time="2020-11-24 22:14:54" level=debug msg="Starting service type *attestations.Service" prefix=registry
time="2020-11-24 22:14:54" level=debug msg="Starting service type *blockchain.Service" prefix=registry
time="2020-11-24 22:14:54" level=debug msg="Starting service type *initialsync.Service" prefix=registry
time="2020-11-24 22:14:54" level=debug msg="Starting service type *sync.Service" prefix=registry
time="2020-11-24 22:14:54" level=debug msg="Starting service type *rpc.Service" prefix=registry
time="2020-11-24 22:14:54" level=debug msg="Starting service type *gateway.Gateway" prefix=registry
time="2020-11-24 22:14:54" level=info msg="Waiting to reach the validator deposit threshold to start the beacon chain..." prefix=blockchain
time="2020-11-24 22:14:54" level=debug msg="Starting service type *prometheus.Service" prefix=registry
time="2020-11-24 22:14:54" level=info msg="Starting JSON-HTTP API" address="127.0.0.1:3500" prefix=gateway
time="2020-11-24 22:14:54" level=info msg="gRPC server listening on port" address="127.0.0.1:4000" prefix=rpc
time="2020-11-24 22:14:54" level=warning msg="You are using an insecure gRPC server. If you are running your beacon node and validator on the same machines, you can ignore this message. If you want to know how to enable secure connections, see: https://docs.prylabs.network/docs/prysm-usage/secure-grpc" prefix=rpc
time="2020-11-24 22:14:54" level=debug msg="Starting prometheus service" address="127.0.0.1:8080" prefix=prometheus
time="2020-11-24 22:14:54" level=error msg="Could not connect to powchain endpoint" error="could not dial eth1 nodes: eth1 node using incorrect chain id, 0 != 1" prefix=powchain

```

**Comment by MAI1O** [comment]
I am experiencing this same issue. Only difference is I'm not using docker. I've tried a number of things but nothing works. Really hoping somebody comes through on this. Sorry I can't contribute anything else.

**Comment by BiggT76** [comment]
I might have a clue:
Just confirmed that the issue appears to be related to how the wallet is created/imported.  I know how two people (myself included) that created the keys on one machine but imported them into another.  Did anyone else here do the same?

Reference:  https://www.reddit.com/r/ethstaker/comments/k0cblo/can_anyone_help_could_not_connect_to_powchain/gdhx69i/?context=3

**Comment by prestonvanloon** [comment]
> I might have a clue:
> Just confirmed that the issue appears to be related to how the wallet is created/imported. I know how two people (myself included) that created the keys on one machine but imported them into another. Did anyone else here do the same?
> 
> Reference: https://www.reddit.com/r/ethstaker/comments/k0cblo/can_anyone_help_could_not_connect_to_powchain/gdhx69i/?context=3

This specific error doesn't have anything to do with importing validator wallets.

For some reason, your eth1 node is returning a chain ID of 0, which doesn't make much sense. ETH1 mainnet chain ID is 1. 

Could you kindly run this and paste the results?
```
curl -X POST --data '{"jsonrpc":"2.0","method":"net_version","params":[],"id":67}' http://192.168.178.170:8545
```

You should see
```json
{"jsonrpc":"2.0","id":67,"result":"1"}
```

The result is the chain ID.

**Comment by BiggT76** [comment]
`invalid content type, only application/json is supported`

**Comment by prestonvanloon** [comment]
@BiggT76, sorry, I edited it. Please try again

**Comment by BiggT76** [comment]
`{"jsonrpc":"2.0","id":67,"result":"0x0"}
`

**Comment by BiggT76** [comment]
Beginning of geth config:

**Comment by prestonvanloon** [comment]
OK, we have identified the issue. Your geth node is not fully synced and for some reason is returning `0x0` for the chain ID.

**Comment by BiggT76** [comment]
That's strange...I'll double check that.  I swore it was updated.
Thanks for your help @prestonvanloon , if the problem persists after full sync I'll provide an update.

**Comment by prestonvanloon** [comment]
@BiggT76 No problem at all! I'm going to see if we can file an issue in go-ethereum.

**Comment by prestonvanloon** [comment]
See: https://github.com/ethereum/go-ethereum/issues/20894

**Comment by prestonvanloon** [comment]
It should be fixed by https://github.com/ethereum/go-ethereum/pull/21686

**Comment by egk10** [comment]
my geth is synced whith a 1 block delay comparing on etherscan. is this consider UNSYNC ???
![sync](https://user-images.githubusercontent.com/69823897/100170018-858ba300-2ea3-11eb-9668-596235df51b2.PNG)

**Comment by BiggT76** [comment]
@egk10 It's my understanding that its still processing all of the _state entries_ records that keep popping up.  This was what I found on the [geth FAQ page](https://geth.ethereum.org/docs/faq):

>  You are just done with the block download phase and still running the state downloads. You can see this yourself via the seemingly endless Imported state entries [...] stream of logs. You’ll need to wait that out too before your node comes truly online.

**Comment by karalabe** [comment]
@prestonvanloon This is not a bug. As long as the node only has the genesis block available, the chain ID is for all intents and purposes `0`, because the notion of a chain id did not exist. Whether we should change the behavior is debatable (and debated a lot really), but it's not a bug, it's intended to work like this.

**Comment by victorelec14** [comment]
> curl -X POST --data '{"jsonrpc":"2.0","method":"eth_chainId","params":[],"id":67}' -H "Content-Type:application/json" http://192.168.178.170:8545
`{"jsonrpc":"2.0","id":67,"result":"0x0"}`



Get Status:

```
INFO [11-25|08:28:29.823] Imported new state entries               count=1465 elapsed=6.439ms      processed=89491081 pending=330673 trieretry=0   coderetry=0  duplicate=7083 unexpected=28991
INFO [11-25|08:28:29.924] Imported new state entries               count=1569 elapsed=12.733ms     processed=89492650 pending=330549 trieretry=0   coderetry=0  duplicate=7083 unexpected=28991
INFO [11-25|08:28:29.982] Imported new state entries               count=1372 elapsed=6.385ms      processed=89494022 pending=331096 trieretry=0   coderetry=0  duplicate=7083 unexpected=28991
INFO [11-25|08:28:30.113] Imported new state entries               count=2020 elapsed=10.194ms     processed=89496042 pending=329624 trieretry=0   coderetry=0  duplicate=7083 unexpected=28991
INFO [11-25|08:28:30.261] Imported new state entries               count=1931 elapsed=10.348ms     processed=89497973 pending=328263 trieretry=0   coderetry=0  duplicate=7083 unexpected=28991
INFO [11-25|08:28:30.351] Imported new state entries               count=1278 elapsed=6.052ms      processed=89499251 pending=328677 trieretry=0   coderetry=0  duplicate=7083 unexpected=28991
INFO [11-25|08:28:30.483] Imported new state entries               count=1862 elapsed=9.003ms      processed=89501113 pending=328354 trieretry=0   coderetry=0  duplicate=7083 unexpected=28991
INFO [11-25|08:28:30.544] Imported new state entries               count=1536 elapsed=6.369ms      processed=89502649 pending=330077 trieretry=0   coderetry=0  duplicate=7083 unexpected=28991
INFO [11-25|08:28:30.636] Imported new state entries               count=1189 elapsed=5.435ms      processed=89503838 pending=331136 trieretry=0   coderetry=0  duplicate=7083 unexpected=28991
INFO [11-25|08:28:30.709] Imported new state entries               count=1536 elapsed=7.336ms      processed=89505374 pending=331643 trieretry=0   coderetry=0  duplicate=7083 unexpected=28991
```

it seems to me that there is something wrong with geht? , it is the second time I synchronize the chain

**Comment by karalabe** [comment]
@victorelec14 There are 650 million state entries. You'll need to wait it out

**Comment by victorelec14** [comment]
@prestonvanloon  @karalabe 

What node do you recommend me to use?
- go-ethereum (Geth)
- parity
- openethereum
- nethermind

**Comment by victorelec14** [comment]
confirmed, I'm on the same BlockNumber as Etherscan, but the "state entries" haven't finished importing.

Maybe you could put an info message in Prism or check the status of the chain and show it?

thanks

**Comment by karalabe** [comment]
@victorelec14 You are **not** on the same block. As long as the states didn't download all the states, you are essentially on block 0. It's a painful process and takes a lot of time, sorry about that. There's a new sync protocol that will solve this, but it's not something served yet by nodes, so it won't help you yet.

**Comment by nisdas** [comment]
@victorelec14 Keep a lookout for this PR 
https://github.com/prysmaticlabs/prysm/pull/7895

It will be merged in the next release. Unfortunately the current error log is confusing as you have noticed. With this PR it will inform you first if your node is 'synced' or not.  We performed the sync check later on in the initial startup routine, which as we have realised might confuse users here when validating the chain id.
